cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java

```diff
@@ -278,26 +285,30 @@ public void onMsgCancel(Integer choiceId) {
         consentFinished();
     }
 
-    public void onMsgRejected(Integer choiceId) {
+    public void onMsgRejected(Integer choiceId) throws ConsentLibException {
         closeAllViews();
         sendConsent(ActionTypes.MSG_REJECT, choiceId);
     }
 
-    public void onMsgShowOptions() {
+    public void onMsgShowOptions() throws ConsentLibException {
         loadPm();
     }
 
-    private void loadPm() {
+    private void loadPm() throws ConsentLibException{
         isPmOn = true;
         loadConsentUI(pmUrl());
     }
 
-    private void loadConsentUI(String url) {
-        mCountDownTimer.start();
-        runOnLiveActivityUIThread(() -> {
-            if (webView == null) webView = buildWebView();
-            webView.loadConsentUIFromUrl(url);
-        });
+    private void loadConsentUI(String url) throws  ConsentLibException {
+        if (!hasLostInternetConnection()) {
```

Comment:
here we can also throw instead of checking having a else clause

Reply:
will do this and follow same guidelines for future tasks